sdk/main/include/ContractUpdateTransaction.h

```diff
@@ -46,286 +40,301 @@ class TransactionBody;
-   * Extract the staked node ID.
+   * Derived from Executable. Submit this ContractUpdateTransaction to a Node.
    *
-   * @return The staked node ID.
+   * @param client   The Client submitting this ContractUpdateTransaction.
+   * @param deadline The deadline for submitting this ContractUpdateTransaction.
+   * @param node     Pointer to the Node to which this ContractUpdateTransaction should be submitted.
+   * @param response Pointer to the TransactionResponse protobuf object that gRPC should populate with the response
+   *                 information from the gRPC server.
+   * @return The gRPC status of the submission.
    */
-  inline InitType<int64_t> getStakedNodeId() { return mStakedNodeId; }
+  [[nodiscard]] grpc::Status submitRequest(const Client& client,
+                                           const std::chrono::system_clock::time_point& deadline,
+                                           const std::shared_ptr<internal::Node>& node,
+                                           proto::TransactionResponse* response) const override;
 
   /**
-   * Extract the staking rewards policy.
+   * Build a ContractUpdateTransactionBody protobuf object from this ContractUpdateTransaction object.
    *
-   * @return \c TRUE if this update is set to decline rewards, otherwise
-   *         \c FALSE.
-   */
-  inline InitType<bool> getDeclineRewards() { return mDeclineReward; }
-
-private:
-  /**
-   * Initialize from the transaction body.
+   * @return A pointer to a ContractUpdateTransactionBody protobuf object filled with this ContractUpdateTransaction
+   *         object's data.
    */
-  void initFromTransactionBody();
+  [[nodiscard]] proto::ContractUpdateTransactionBody* build() const;
 
   /**
-   * The ID of the contract to be updated.
+   * The ID of the contract to update.
    */
-  InitType<ContractId> mContractId;
+  ContractId mContractId;
 
   /**
-   * The new expiry of the contract, no earlier than the current expiry
-   * (resolves to EXPIRATION_REDUCTION_NOT_ALLOWED otherwise).
+   * The new expiration time for the contract.
    */
-  InitType<std::chrono::nanoseconds> mExpirationTime;
+  std::optional<std::chrono::system_clock::time_point> mExpirationTime;
 
   /**
-   * The new key to control updates to the contract.
+   * The new admin key to be used for the contract.
    */
-  std::shared_ptr<PublicKey> mAdminKey;
+  std::shared_ptr<PublicKey> mAdminKey = nullptr;
 
   /**
-   * (NOT YET IMPLEMENTED) The new interval at which the contract will pay to
-   * extend its expiry (by the same interval).
+   * The new auto renew period for the contract.
    */
-  InitType<std::chrono::seconds> mAutoRenewPeriod;
+  std::optional<std::chrono::duration<double>> mAutoRenewPeriod;
 
   /**
-   * The new contract memo, assumed to be Unicode encoded with UTF-8 (at most
-   * 100 bytes)
+   * The new memo for the contract (UTF-8 encoding max 100 bytes).
    */
-  InitType<std::string> mMemo;
+  std::optional<std::string> mMemo;
```

Comment:
My request is not related to the current change, but I noticed field `contractMemo` is named contractMemo in [Java SDK](https://github.com/hashgraph/hedera-sdk-java) version of file [ContractUpdateTransaction](https://github.com/hashgraph/hedera-sdk-java/blob/develop/sdk/src/main/java/com/hedera/hashgraph/sdk/ContractUpdateTransaction.java).

Can we rename it to `mContractMemo` to keep the same naming convention between the SDKs?